src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node currentNode = table[getIndex((key))];
+        Node<K, V> newNode = new Node<>(Objects.hashCode(key), key, value, null);
+        if (currentNode == null) {
+            table[getIndex((key))] = newNode;
+            size++;
+        }
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                currentNode.value = value;
+                break;
+            } else if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+            }
+            currentNode = currentNode.next;
+        }
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return (V) currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(Object key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        size = 0;
+        threshold = table.length << 1;
+        Node[] oldTable = table;
+        table = new Node[table.length << 1];
+        threshold = (int) (table.length * LOAD_FACTOR);
+        transfer(oldTable);
+    }
+
+    private void transfer(Node<K, V>[] table) {
+        for (Node<K, V> kvNode : table) {
+            while (kvNode != null) {
+                put(kvNode.key, kvNode.value);
+                kvNode = kvNode.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
+
+        private final K key;
+        private V value;
+
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K,V> next) {
```

Comment:
where do you declare and initialize variable int hash?

Reply:
int hash has been removed